contracts/polyx-vesting/lib.rs

```diff
@@ -0,0 +1,253 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang as ink;
+
+#[ink::contract]
+mod polyx_vesting {
+    use ink_storage::traits::SpreadAllocate;
+
+    /// Defines the storage of your contract.
+    #[ink(storage)]
+    #[derive(SpreadAllocate)]
+    pub struct PolyxVesting {
+        released: Balance,
+        beneficiary: AccountId,
+        start: Timestamp,
+        duration: Timestamp,
+    }
+
+    /// Event emitted when Polyx is released.
+    #[ink(event)]
+    pub struct PolyxReleased {
+        value: Balance,
+    }
+
+    /// The error types.
+    #[derive(Debug, PartialEq, Eq, scale::Encode, scale::Decode)]
+    #[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+    pub enum Error {
+        /// Returned if not enough balance to fulfill a request is available.
+        InsufficientBalance,
+        /// Funds are not released as yet.
+        FundsNotReleased,
+        /// Invalid start timestamp
+        InvalidStartTimestamp,
+        /// Duration Overflow
+        DurationOverflow,
+        /// Invalid timestamp
+        InvalidTimestamp,
+        /// Timestamp Overflow
+        TimestampOverflow,
+    }
+
+    /// The contract result type.
+    pub type Result<T> = core::result::Result<T, Error>;
+
+    impl PolyxVesting {
+        /// Constructor
+        #[ink(constructor, payable)]
+        pub fn new(
+            beneficiary_address: AccountId,
+            start_timestamp: Timestamp,
+            duration_milli_seconds: Timestamp,
+        ) -> Result<Self> {
+            ink_lang::utils::initialize_contract(|contract| {
+                Self::new_init(
+                    contract,
+                    beneficiary_address,
+                    start_timestamp,
+                    duration_milli_seconds,
+                )
+            })
+        }
+
+        fn new_init(
+            &mut self,
+            beneficiary_address: AccountId,
+            start_timestamp: Timestamp,
+            duration_milli_seconds: Timestamp,
+        ) -> Result<()> {
+            self.beneficiary = beneficiary_address;
+            self.start = start_timestamp;
+            self.duration = duration_milli_seconds;
+            self.validation()
+        }
+
+        fn validation(&self) -> Result<()> {
+            // Ensure start isn't zero
+            if self.start == 0 || self.duration == 0 {
```

Comment:
There is already a check to make sure that `start` isn't less than the current block time.  No need to check that it isn't zero.

Also I think it might be ok to allow a zero duration.  If the vesting schedule just wants to use the `start` time to release the total amount.